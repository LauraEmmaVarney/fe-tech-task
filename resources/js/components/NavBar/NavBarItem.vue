<template>
    <div class="col-4 header-checkout-step animated fadeInDown faster">
        <div class="row align-items-center">
            <div class="col-md-5 text-center text-md-right">
                <i 
                    :class="[
                        'border border-3 rounded-circle p-3 fal fa-2x', 
                        icon,
                        isActiveItem ? 'border-primary text-primary' : 'border-secondary text-secondary'
                    ]"
                />
            </div>
            <div class="col-md-7 text-center text-md-left">
                <span 
                    :class="[
                        'mt-2 mt-md-0 badge small font-weight-normal text-uppercase px-0',
                        isActiveItem ? 'text-primary' : 'text-secondary'
                    ]" 
                >
                    Step {{ step }}
                </span>
                <p class="mb-0 d-none d-md-block">{{ label }}</p>
            </div>
        </div>
    </div>
</template>

<script>
    export default {
        name: "NavBarItem",
        props: {
            label: {
                type: String,
                required: true,
            },
            step: {
                type: Number,
                required: true,
            },
            icon: {
                type: String,
                required: true,
            },
            activeStep: {
                type: Number,
                required: true,
            }
        },
        computed: {
            isActiveItem: function() {
                return this.activeStep === this.step
            }
        },
    }

</script>

<style scoped>

</style>
