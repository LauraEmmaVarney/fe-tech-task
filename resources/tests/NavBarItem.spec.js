import { shallowMount } from '@vue/test-utils'
import NavBarItem from '../js/components/NavBar/NavBarItem.vue';

describe('NavBarItem component tests', () => {
    const wrapper = shallowMount(NavBarItem, {
        propsData: {
            activeStep: 1,
            icon: 'fa-user-circle',
            label: 'Your Details',
            step: 1
        }
    });

    it('mounts the component', () => {
        expect(wrapper.exists()).toBe(true);
    });

    it('checks props data', () => {
        expect(wrapper.props().activeStep).toBe(1);
        expect(wrapper.props().icon).toBe('fa-user-circle');
        expect(wrapper.props().label).toBe('Your Details');
        expect(wrapper.props().step).toBe(1);
    });

    it('checks computed properties', () => {
        expect(wrapper.vm.isActiveItem).toBe(true);
    });

    it('check against jest snapshot test', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
});