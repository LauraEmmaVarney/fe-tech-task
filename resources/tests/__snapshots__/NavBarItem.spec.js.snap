// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarItem component tests check against jest snapshot test 1`] = `
<div
  class="col-4 header-checkout-step animated fadeInDown faster"
>
  <div
    class="row align-items-center"
  >
    <div
      class="col-md-5 text-center text-md-right"
    >
      <i
        class="border border-3 rounded-circle p-3 fal fa-2x fa-user-circle border-primary text-primary"
      />
    </div>
     
    <div
      class="col-md-7 text-center text-md-left"
    >
      <span
        class="mt-2 mt-md-0 badge small font-weight-normal text-uppercase px-0 text-primary"
      >
        
                Step 1
            
      </span>
       
      <p
        class="mb-0 d-none d-md-block"
      >
        Your Details
      </p>
    </div>
  </div>
</div>
`;
